File: src/com/company/v2/릿코드/알고리즘/medium/_53.java
package com.company.v2.릿코드.알고리즘.medium;

public class _53 {

    public int maxSubArray(int[] nums) {
        int result = nums[0];
        int[] dp = new int[nums.length];
        dp[0] = nums[0];

        for (int i = 1; i < nums.length; i++) {
            dp[i] = Math.max(dp[i - 1] + nums[i], nums[i]);
            result = Math.max(result, dp[i]);
        }

        return result;
    }
}

File: src/com/company/v2/릿코드/알고리즘/medium/_53.kt
package com.company.v2.릿코드.알고리즘.medium

import kotlin.math.max

fun maxSubArray(nums: IntArray): Int {
    var result: Int = nums[0]
    val dp = IntArray(nums.size)
    dp[0] = nums[0]

    val sorted = listOf(1, 2).sorted()

    for (i: Int in 1 until nums.size) {
        dp[i] = max(dp[i-1] + nums[i], nums[i])
        result = max(result, dp[i])
    }

    return result
}

fun main(){
    println(maxSubArray(IntArray(1) { 1 }))
}